admin/securedrop_admin/__init__.py

```diff
@@ -667,29 +668,28 @@ def update(args):
 
     sdlog.info("Verifying signature on latest update...")
 
-    try:
-        # First try to get the release key using Tails default keyserver
-        get_release_key_from_keyserver(args)
-    except subprocess.CalledProcessError:
-        # Now try to get the key from a secondary keyserver.
-        secondary_keyserver = 'hkps://hkps.pool.sks-keyservers.net'
-        get_release_key_from_keyserver(args,
-                                       keyserver=secondary_keyserver)
+    # Retrieve key from openpgp.org keyserver
+    get_release_key_from_keyserver(args,
+                                   keyserver=DEFAULT_KEYSERVER)
 
     git_verify_tag_cmd = ['git', 'tag', '-v', latest_tag]
     try:
         sig_result = subprocess.check_output(git_verify_tag_cmd,
                                              stderr=subprocess.STDOUT,
                                              cwd=args.root)
 
-        good_sig_text = 'Good signature from "SecureDrop Release Signing Key"'
+        good_sig_text = ['Good signature from "SecureDrop Release Signing ' +
+                         'Key"',
+                         'Good signature from "SecureDrop Release Signing ' +
+                         'Key <securedrop-release-key@freedom.press>"']
         bad_sig_text = 'BAD signature'
         # To ensure that an adversary cannot name a malicious key good_sig_text
         # we check that bad_sig_text does not appear and that the release key
         # appears on the second line of the output.
         gpg_lines = sig_result.split('\n')
         if RELEASE_KEY in gpg_lines[1] and \
-                sig_result.count(good_sig_text) == 1 and \
+                len([s for s in gpg_lines if
```

Comment:
To me, these two new lines introduced appear complex, especially for such a critical part of the verification code. Do you think it makes sense to clarify the logic here (either through comments or making the logic more explicit)?

Reply:
I reorganized things to make it a little clearer and added comments.